### PR TITLE
feat (jkube-kit/config/service) : Add BuildPackBuildService (#2493)

### LINKED ISSUE #439: [EPIC] Buildpacks-based container image build strategy
### Description

As a user, I want to be able to use Eclipse JKube to build container images and deploy them to a Kubernets-compatible cluster for my Java application leveraging Buildpacks, so that I can take advantage and reuse one of the many production ready [buildpacks](https://registry.buildpacks.io/) to create an image for my application.

Buildpacks is used by many frameworks, such as Spring Boot, to create container images for an application. There is a curated, community-maintained, growing list of buildpacks to support all kinds of frameworks, libraries, and programming languages.

Currently, [JKube can be made compatible](https://www.callibrity.com/blog/spring-into-kubernetes) with Buildpacks-produced container images by tuning the image name at the buildpacks level, or modifying JKube's resource configuration.
However, this is cumbersome and requires the user to provide additional configurations.

It would be good to have a transparent, seamless integration with Buildpacks (similar to what we have for Jib). This way, a user wouldn't need to worry about additional configurations and could simply leverage Buildpacks by configuring it as JKube's [container image build strategy](https://eclipse.dev/jkube/docs/kubernetes-maven-plugin/#build-goal-configuration).

### Tasks
- [x] Spike a PoC implementation to see if this is possible
- [x] Draft a design document
  https://docs.google.com/document/d/1HYSEad6pLBvtHJqyCmNdUUTE3lhIdCDd1AtmA5p-9MY/
- [x] Create individual development tasks and stories
  - https://github.com/eclipse/jkube/issues/2465
    - https://github.com/eclipse/jkube/issues/2453
    - https://github.com/eclipse/jkube/issues/2454
      - https://github.com/eclipse/jkube/issues/2456
    - #2493
    - #2491
    - #2492 
    - #2464
  - https://github.com/eclipse/jkube/issues/2470
  - https://github.com/eclipse/jkube/issues/2459
  - https://github.com/eclipse/jkube/issues/2466
  - #2494
    - https://github.com/eclipse/jkube/issues/2461
    - https://github.com/eclipse/jkube/issues/2462
  - https://github.com/eclipse-jkube/jkube/issues/3097 
  
To be discussed later:
  - https://github.com/eclipse/jkube/issues/2460

  - https://github.com/eclipse/jkube/issues/2463

### See also:
- https://buildpacks.io/
- https://github.com/paketo-buildpacks
- https://github.com/spring-projects/spring-boot/tree/3657c6c85de0bd49529d2dc256dd4350b61ad7c1/spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform

### Relates to:
- #66

### LINKED ISSUE #2493: Buildpacks build service
### Component

None

### Description
As a JKube developer, I want to be able to invoke the pack cli through a JKube service to configure and run a build, So i can bind that to the JKube's plugins or just with the Java api, in a central place.

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

- [x] Requires https://github.com/eclipse/jkube/pull/2496 https://github.com/eclipse/jkube/pull/2498 to be merged first

Related to https://github.com/eclipse/jkube/issues/2465

Fixes #2493

+ Add new build strategy called `buildpacks` as new enum for buildpacks in JKubeBuildStrategy
+ Add BuildPackBuildService that would be activated when build strategy
      is set to buildpacks

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2448 (_2023-11-16T06:22:34Z_)

:gear: [JKube E2E Tests (6887114175)](https://github.com/jkubeio/ci/actions/runs/6887114175) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2448)](https://github.com/jkubeio/ci/actions/runs/6887114175/job/18733810533)

**Comment by manusa** [comment]
>  I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged

This is not small and simple.

**Comment by rohanKanojia** [comment]
There are only 24 files. I'll add documentation in a follow up PR.

File: jkube-kit/build/service/buildpacks/pom.xml [hunk 1]
```diff
@@ -74,6 +74,18 @@
...
+        <artifactId>maven-jar-plugin</artifactId>
+        <executions>
+          <execution>
+            <goals>
+              <goal>test-jar</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/build/service/buildpacks/pom.xml:89]: No

File: jkube-kit/build/service/buildpacks/src/test/java/org/eclipse/jkube/kit/service/buildpacks/AbstractBuildPackCliDownloaderTest.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ abstract class AbstractBuildPackCliDownloaderTest {
   abstract String getProcessorArchitecture();
 
   @BeforeEach
-  void setUp() {
+  void setUp() throws IOException {
```
Comment (manusa) [inline jkube-kit/build/service/buildpacks/src/test/java/org/eclipse/jkube/kit/service/buildpacks/AbstractBuildPackCliDownloaderTest.java:58]: I think this is also unneeded now

**Review by manusa — APPROVED** [review]

**Pull request merged**